test/0.8.25/Accounting.t.sol

```diff
@@ -0,0 +1,360 @@
+
+    function getPooledEthByShares(uint256 _sharesAmount) external view returns (uint256);
+
+    function resume() external;
+
+    function getBeaconStat()
+        external
+        view
+        returns (uint256 depositedValidators, uint256 beaconValidators, uint256 beaconBalance);
+}
+
+interface ISecondOpinionOracleMock {
+    function mock__setReportValues(
+        bool _success,
+        uint256 _clBalanceGwei,
+        uint256 _withdrawalVaultBalanceWei,
+        uint256 _totalDepositedValidators,
+        uint256 _totalExitedValidators
+    ) external;
+}
+
+// 0.002792 * 10^18
+// 0.0073 * 10^18
+uint256 constant maxYiedPerOperatorWei = 2_792_000_000_000_000; // which % of slashing could be?
+uint256 constant maxLossPerOperatorWei = 7_300_000_000_000_000;
+uint256 constant stableBalanceWei = 32 * 1 ether;
+
+struct FuzzValues {
+    uint256 _preClValidators;
+    uint256 _preClBalanceWei;
+    uint256 _clValidators;
+    uint256 _clBalanceWei;
+    uint256 _withdrawalVaultBalance;
+    uint256 _elRewardsVaultBalanceWei;
+    uint256 _sharesRequestedToBurn;
+    uint256 _lidoExecutionLayerRewardVaultWei;
+}
+
+struct LidoTransfer {
+    address from;
+    address to;
+}
+
+contract AccountingHandler is CommonBase, StdCheats, StdUtils {
+    struct Ghost {
+        int256 clValidators;
+        int256 depositedValidators;
+        int256 sharesMintAsFees;
+        int256 transferShares;
+        int256 totalRewardsWei;
+        int256 principalClBalanceWei;
+        int256 unifiedClBalanceWei;
+    }
+
+    IAccounting private accounting;
+    ILido private lido;
+    ISecondOpinionOracleMock private secondOpinionOracle;
+    IStakingRouter public stakingRouter;
+
+    Ghost public ghost;
+    LidoTransfer[] public ghost_lidoTransfers;
+
+    address private accountingOracle;
+    address private lidoExecutionLayerRewardVault;
+    address private burner;
+    LimitsList public limitList;
+
+    constructor(
+        address _accounting,
+        address _lido,
+        address _accountingOracle,
+        LimitsList memory _limitList,
+        address _lidoExecutionLayerRewardVault,
+        address _secondOpinionOracle,
+        address _burnerAddress,
+        address _stakingRouter
+    ) {
+        accounting = IAccounting(_accounting);
+        lido = ILido(_lido);
+        accountingOracle = _accountingOracle;
+        limitList = _limitList;
+        lidoExecutionLayerRewardVault = _lidoExecutionLayerRewardVault;
+
+        ghost = Ghost(0, 0, 0, 0, 0, 0, 0);
+        secondOpinionOracle = ISecondOpinionOracleMock(_secondOpinionOracle);
+        burner = _burnerAddress;
+        stakingRouter = IStakingRouter(_stakingRouter);
+    }
+
+    function cutGwei(uint256 value) public returns (uint256) {
+        return (value / 1 gwei) * 1 gwei;
+    }
+
+    function handleOracleReport(FuzzValues memory fuzz) external {
+        uint256 _timeElapsed = 86_400;
+        uint256 _timestamp = 1_737_366_566 + _timeElapsed;
```

Comment:
Why do we need a specific timestamp? Can block.timestamp be used?

Reply:
replaced